modules/gdscript/gdscript.cpp

```diff
@@ -1649,6 +1649,17 @@ void GDScriptLanguage::reload_tool_script(const Ref<Script> &p_script, bool p_so
 	for (Map<Ref<GDScript>, Map<ObjectID, List<Pair<StringName, Variant> > > >::Element *E = to_reload.front(); E; E = E->next()) {
 
 		Ref<GDScript> scr = E->key();
+		Ref<Script> rel_scr = ResourceLoader::load(scr->get_path(), scr->get_class(), true);
```

Comment:
This may break scripts that were edited in the built-in editor but not saved? I think there has to be a different way to do this.

Reply:
This is still unresolved @T4g1.

Reply:
Sorry, I'm finally back into gamedev, I'll work on that this week.

Reply:
I looked at it, this fix is no longer working and I don't have a good enough understanding of the engine to fix it right now. I doubt that I will dive more on this part as I do not use an external editor anymore but here is where I'm stuck: https://github.com/godotengine/godot/blob/master/editor/plugins/script_editor_plugin.cpp#L785 So we keep the edited script and we then want to script->reload() it, problem is, "has_instances" is true in GDScript::reload so the fail condition triggers: What instances does and why it needs to be empty for reload is unclear for me but depending on that, one could: a) discard it before reload, b) clear it and re-populate it after reload, c) remove the fail condition on has_instances